Add delete button on palette card

mohitkyadav opened the pull request:
Fixes #20 

<img width="1165" alt="Screenshot 2022-06-22 at 7 11 01 AM" src="https://user-images.githubusercontent.com/25580776/174948517-be03e651-6db8-417e-9e94-71af9fdb2c50.png">



## CHANGELOG
### Added
- Delete icon button on the palette card.
- Palette card title now truncates.
- All cards persist same height.

mohitkyadav commented:
@colebemis please review.

mohitkyadav commented:
@colebemis still waiting over here.

colebemis commented:
Thanks for your patience, @mohitkyadav! Hoping to find time to merge this today

colebemis reviewed (commented):
This looks great, @mohitkyadav! ✨ 

A couple of notes:
* I chatted with some of our accessibility experts and they discouraged us from nesting a `<button>` inside a `<a>`. We can achieve a similar visual layout without nesting the markup by making the `<button>` sibling of the `<a>` and absolutely positioning the `<button>` inside the `<a>`. 
* Can we make the buttons match the background of the palette? Similar to the buttons on each palette page:

https://user-images.githubusercontent.com/4608155/174847866-9ba688ba-ecc3-43ce-88f6-3b0345c3a8e0.mp4

Let me know if you have any clarifying questions :)

stevenpetryk commented on src/pages/index.tsx:
```diff
@@ -115,8 +116,32 @@ export function Index(props: RouteComponentProps) {
+                        0.85),
+                    }
+                  }}
+                >Delete alette</IconButton>
```
This is a typo, but: do you need this inner text _and_ the aria-label? Or just one?

stevenpetryk commented:
I think this will cause prettier to fail on `main` btw—it was opened before #22 was merged so Prettier isn't running in CI yet.

stevenpetryk commented on src/pages/index.tsx:
```diff
@@ -55,26 +55,27 @@ export function Index(props: RouteComponentProps) {
             gap: 3,
-            gridTemplateColumns: 'repeat(auto-fill, minmax(300px, 1fr))'
+            gridTemplateColumns: 'repeat(auto-fill, minmax(300px, 1fr))',
+            gridTemplateRows: '230px'
```
Could probably get away with [`line-clamp`](https://caniuse.com/css-line-clamp) instead of doing this.

mohitkyadav replied:
I didn't know it could be used like that. Thank you 👏 .

stevenpetryk commented on src/pages/index.tsx:
```diff
@@ -116,7 +117,13 @@ export function Index(props: RouteComponentProps) {
+                <Text sx={{
+                  lineHeight: '1',
+                  textOverflow: 'ellipsis',
+                  width: '80%',
```
Doesn't a percentage width here mean there will be a variable amount of space between the text and the button?

mohitkyadav replied:
Since the grid items don't have fixed width (`gridTemplateColumns: 'repeat(auto-fill, minmax(300px, 1fr))',`), the text also does not have a fixed width, I just put 80% to be on the safe side. Please correct me if I'm wrong.

mohitkyadav replied:
I think I missed your point, I don't know if it's a bad practice to have the spacing between the text and button change. Here I put 80% to avoid the button covering the text. Wdyt?

mohitkyadav commented:
1. Updated the colour of the delete button.
2. Moved the button outside of the card.
3. palette title now truncates.
4. All cards persist same height.

<img width="1165" alt="Screenshot 2022-06-22 at 7 05 58 AM" src="https://user-images.githubusercontent.com/25580776/174947832-efe2040c-92df-483f-bfca-2aae1be75d42.png">

mohitkyadav commented on src/pages/index.tsx:
```diff
@@ -115,8 +117,36 @@ export function Index(props: RouteComponentProps) {
-                <Text sx={{lineHeight: '1'}}>{palette.name}</Text>
+                <Text
+                  sx={{
+                    lineHeight: '1.2',
```
Had to do this to fix the bottom part cutting as visible in this screenshot. Please let me know if there's another way.

<img width="1165" alt="Screenshot 2022-06-22 at 7 05 58 AM" src="https://user-images.githubusercontent.com/25580776/174948631-65d38fc0-0b16-4f0d-8f87-dbed11859f44.png">

colebemis commented on src/pages/index.tsx:
```diff
@@ -115,8 +117,36 @@ export function Index(props: RouteComponentProps) {
+                  {palette.name}
+                </Text>
               </Box>
+              <IconButton
```
We could reuse the same IconButton component that we use on the palette page here. 

Here's how I'd do it:

```diff
diff --git a/src/pages/index.tsx b/src/pages/index.tsx
index f6ef37d..0914aba 100644
--- a/src/pages/index.tsx
+++ b/src/pages/index.tsx
@@ -1,8 +1,17 @@
 import {LinkExternalIcon, MarkGithubIcon, PlusIcon, TrashIcon} from '@primer/octicons-react'
-import {Box, Button, Heading, IconButton, Label, Link as PrimerLink, StyledOcticon, Text} from '@primer/react'
+import {
+  Box,
+  Button,
+  Heading,
+  IconButton as PrimerIconButton,
+  Label,
+  Link as PrimerLink,
+  StyledOcticon,
+  Text
+} from '@primer/react'
 import {Link, RouteComponentProps} from '@reach/router'
 import {mix, readableColor} from 'color2k'
-import React from 'react'
+import {IconButton} from '../components/button'
 import {routePrefix} from '../constants'
 import {useGlobalState} from '../global-state'
 import {colorToHex, getColor} from '../utils'
@@ -38,7 +47,7 @@ export function Index(props: RouteComponentProps) {
             GitHub
             <StyledOcticon icon={LinkExternalIcon} sx={{marginLeft: 1}} />
           </PrimerLink>
-          <IconButton
+          <PrimerIconButton
             aria-label="Create new palette"
             icon={PlusIcon}
             onClick={() => send('CREATE_PALETTE')}
@@ -119,7 +128,7 @@ export function Index(props: RouteComponentProps) {
                 </Box>
                 <Text
                   sx={{
-                    lineHeight: '1.2',
+                    lineHeight: '24px',
                     textOverflow: 'ellipsis',
                     width: '80%',
                     overflow: 'hidden',
@@ -136,15 +145,22 @@ export function Index(props: RouteComponentProps) {
                   send({type: 'DELETE_PALETTE', paletteId: palette.id})
                 }}
                 sx={{
-                  color: readableColor(palette.backgroundColor),
-                  backgroundColor: palette.backgroundColor,
-                  borderColor: mix(readableColor(palette.backgroundColor), palette.backgroundColor, 0.75),
+                  '--color-text': readableColor(palette.backgroundColor),
+                  '--color-background': palette.backgroundColor,
+                  '--color-background-secondary': mix(
+                    readableColor(palette.backgroundColor),
+                    palette.backgroundColor,
+                    0.9
+                  ),
+                  '--color-background-secondary-hover': mix(
+                    readableColor(palette.backgroundColor),
+                    palette.backgroundColor,
+                    0.85
+                  ),
+                  '--color-border': mix(readableColor(palette.backgroundColor), palette.backgroundColor, 0.75),
                   position: 'absolute',
                   right: 3,
-                  bottom: 2,
-                  '&:hover:not([disabled])': {
-                    backgroundColor: mix(readableColor(palette.backgroundColor), palette.backgroundColor, 0.85)
-                  }
+                  bottom: '12px'
                 }}
               />
             </li>
```

mohitkyadav replied:
Updated

colebemis commented on src/pages/index.tsx:
```diff
@@ -115,8 +126,43 @@ export function Index(props: RouteComponentProps) {
+                </Text>
               </Box>
+              <IconButton
+                aria-label="Delete Palette"
```
```suggestion
                aria-label="Delete palette"
```

colebemis reviewed (approved):
Love it. Thank you, @mohitkyadav! 💖
